src/modules/authorizer/role/AUT_TokenGated_Roles_v1.sol

```diff
@@ -89,42 +89,89 @@ contract AUT_TokenGated_Roles_v1 is IAUT_TokenGated_Roles_v1, AUT_Roles_v1 {
     // Stores the threshold amount for each token in a role
     mapping(bytes32 => uint) public thresholdMap;
 
+    // Storage gap for future upgrades
+    uint[50] private __gap;
+
     //--------------------------------------------------------------------------
     // Overloaded and overriden functions
 
-    function hasRole(bytes32 roleId, address account)
+    function hasRole(bytes32 roleId, address who)
         public
         view
         virtual
         override(AccessControlUpgradeable, IAccessControl)
         returns (bool)
     {
         if (isTokenGated[roleId]) {
-            return _hasTokenRole(roleId, account);
+            return _hasTokenRole(roleId, who);
         } else {
-            return super.hasRole(roleId, account);
+            return super.hasRole(roleId, who);
         }
     }
 
     /// @notice Grants a role to an address
     /// @param role The role to grant
-    /// @param account The address to grant the role to
+    /// @param who The address to grant the role to
     /// @return bool Returns if the role has been granted succesful
-    /// @dev Overrides {_grantRole} from AccesControl to enforce interface implementation when role is token-gated
-    function _grantRole(bytes32 role, address account)
+    /// @dev Overrides {_grantRole} from AccessControl to enforce interface implementation and threshold existence when role is token-gated
+    /// @dev Please note: current check for validating a valid token is not conclusive and could be
+    ///         circumvented through a callback() function
+    function _grantRole(bytes32 role, address who)
         internal
         virtual
         override
         returns (bool)
     {
         if (isTokenGated[role]) {
-            try TokenInterface(account).balanceOf(address(this)) {}
-            catch {
-                revert Module__AUT_TokenGated_Roles__InvalidToken(account);
+            // Make sure that a threshold has been set before granting the role
+            if (getThresholdValue(role, who) == 0) {
+                revert Module__AUT_TokenGated_Roles__TokenRoleMustHaveThreshold(
+                    role, who
+                );
+            }
+
+            // Check that address has code attached
```

Comment:
Adding these additional checks indeed are no guarantee. We have added this to the Natspec comment as well. We decided to add them here nonetheless, on recommendations of 0xMacro as it can never hurt.

In relation to doing the checks earlier in the process that makes sense

Reply:
hasTWe have this check at this point since several functions end up calling _grantRole. This way we ensure that if the role is token-gated, it will need to have been configured correctly (by going through grantTokenRoleFromModule). This way you prevent granting a  tokenGated to a particular user addreess

Reply:
Okay so then we would leave it at this location @0xNuggan ?